src/libzerocoin/bignum.h

```diff
@@ -837,9 +837,20 @@ class CBigNum
 
     uint256 getuint256() const
     {
-        uint256 n = 0;
-        mpz_export((unsigned char*)&n, NULL, -1, 1, 0, 0, bn);
-        return n;
+        size_t size = (mpz_sizeinbase (bn, 2) + CHAR_BIT-1) / CHAR_BIT;
+        // This is just a fall back for invalid numbers, it will only take the last 256 bits as openssl implementation does.
+        if(size > 32){
+            std::vector<unsigned char> vch(size);
+            mpz_export(&vch[0], NULL, -1, 1, 0, 0, bn);
+            uint256 n = 0;
```

Comment:
nit: move the initialization of n before the if and change it to `uint2556 n = uint256(); `